src/System.IO.Pipelines/PipelineWriterExtensions.cs

```diff
@@ -9,7 +9,12 @@ public static class PipelineWriterExtensions
     {
         private readonly static Task _completedTask = Task.FromResult(0);
 
-        public static Task WriteAsync(this IPipeWriter output, Span<byte> source)
+        public static Task WriteAsync(this IPipeWriter output, byte[] source)
+        {
+            return WriteAsync(output, new ArraySegment<byte>(source));
+        }
+
+        public static Task WriteAsync(this IPipeWriter output, ArraySegment<byte> source)
```

Comment:
But it's no very useful because most clients are `async` methods and having this overload causes them to blow up.